src/BaseControllerV2.ts

```diff
@@ -82,19 +82,31 @@ export type StateMetadata<T> = {
 };
 
 /**
- * Metadata for a single state property
- *
- * @property persist - Indicates whether this property should be persisted
- *   (`true` for persistent, `false` for transient), or is set to a function
- *   that derives the persistent state from the state.
- * @property anonymous - Indicates whether this property is already anonymous,
- *   (`true` for anonymous, `false` if it has potential to be personally
- *   identifiable), or is set to a function that returns an anonymized
- *   representation of this state.
+ * Metadata for a single state property.
  */
 export interface StatePropertyMetadata<T> {
+  /**
+   * Indicates whether this property should be persisted (`true` for persistent,
+   * `false` for transient), or is set to a function that derives the persistent
+   * state from this state.
+   */
   persist: boolean | StateDeriver<T>;
+
+  /**
+   * Indicates whether this property is already anonymous, (`true` for
+   * anonymous, `false` if it has potential to be personally identifiable),
+   * or is set to a function that returns an anonymized representation of this
+   * state.
+   */
   anonymous: boolean | StateDeriver<T>;
+
+  /**
+   * Indicates whether this property should be communicated across serialization
+   * boundaries such as `window.postMessage` (`true` if so, `false` if not), or
+   * is set to a function that derives state that should be serialized from this
+   * state.
+   */
+  serialize: boolean | StateDeriver<T>;
```

Comment:
Not sure about the name :thinking: Though I'm having trouble thinking of a better one.

It's not so much that this state is serializable, or that it should be serialized. All of the persisted state gets serialized for example, as part of persistence. Our persistence strategy involves a "serialization boundary" as described here.

It's more like.... do we want this state to be.... part of the overall application state? Or maybe whether we want it to be "hidden" or "internal" (except for persistence). I don't know.

Reply:
Hmm, that makes sense. `local` is better than the other suggestions so far at least! And I guess the `boolean` would be flipped (`true` means it should be "hidden").

This brings up another question - should `getState` return properties that are `local`? Maybe we could add a parameter that returned `local` properties, but exclude them by default. That would make this work without any further changes in either product, and it kinda makes sense as a default, since other controllers shouldn't need to peek at local state. But it still allows the persistence layer to request _all_ state so that it can be run through the `getPersistentState` function.

Reply:
Noting here for the record that Erik later came up with the suggestion to use `public` instead (the opposite of `local` essentially) because it would be slightly less awkward to use the state derivers that way.